stellargraph/layer/watch_your_step.py

```diff
@@ -0,0 +1,199 @@
+import tensorflow as tf
+from tensorflow.keras.layers import Layer, Embedding, Input, Lambda, Concatenate
+from tensorflow.keras import backend as K
+from tensorflow.keras import regularizers, initializers, constraints
+import numpy as np
+
+from ..mapper.adjacency_generators import AdjacencyPowerGenerator
+from ..core.experimental import experimental
+
+
+@experimental(reason="lack of unit tests")
+class AttentiveWalk(Layer):
+    """
+    This implements the graph attention as in Watch Your Step: Learning Node Embeddings via Graph Attention
+    https://arxiv.org/pdf/1710.09599.pdf.
+
+    .. warning::
+
+        This class is experimental: it is insufficiently tested.
+
+    Args:
+        walk_length (int): the length of the random walks. Equivalent to the number of adjacency powers used.
+        attention_initializer (str or func): The initialiser to use for the attention weights;
+            defaults to 'glorot_uniform'.
+        attention_regularizer (str or func): The regulariser to use for the attention weights;
+            defaults to None.
+        attention_constraint (str or func): The constraint to use for the attention weights;
+            defaults to None.
+    """
+
+    def __init__(self, walk_length=5, **kwargs):
```

Comment:
Magic number? I'll change it to the default value of 10 in the paper.

Reply:
I think its pretty standard to have decent default parameter for algorithms (e.g. all the scipy algos have defaults). its also convenient for data scientists and means they can just test things out without having to look into an algorithm.

Reply:
Sounds reasonable. Could you have either a reference in the documentation or just a comment pointing out where the default comes from, so that it's not entirely a magic number?